salt/pillar/makostack.py

```diff
@@ -432,49 +452,57 @@ def ext_pillar(minion_id, pillar, *args, **kwargs):
         else:
             namespace = None
         if not os.path.isfile(cfg):
-            log.warning('Ignoring Stack cfg "%s": file does not exist', cfg)
+            log.warning('Ignoring MakoStack cfg `{}`: file not found'.format(cfg))
             continue
-        stack = _process_stack_cfg(cfg, stack, minion_id, pillar, namespace)
+        stack = _process_stack_cfg(cfg, stack, minion_id, pillar, namespace, config)
     return stack
 
 
-def _process_stack_cfg(cfg, stack, minion_id, pillar, namespace):
+def _process_stack_cfg(cfg, stack, minion_id, pillar, namespace, config):
```

Comment:
The CodeClimate static analyzer is wondering if we might be able to split this function out even further to reduce its length and complexity. Thoughts?

Reply:
I'm looking at it now...  Other than compressing a couple of function calls which happen to be split over several lines - e.g. the two `lookup.get_template()` calls, I don't really see anything that is reasonably factorable.  It's a pretty simple try loop processing lines from a config file, with two (necessary) except clauses, and then a return.  After the above consolidation, the function is 36 LOC, but 8 of THOSE lines are logging, so practically speaking it's 28 LOC.

I don't really see any obvious lines of demarc for decomposing it further, but if you have a suggestion I'll happily implement it :)

Reply:
Totally fine. I just like it to bring it to the attention of contributors. As with all static analyzers, sometimes it makes sense and sometimes it doesn't. Thanks as always, @tkwilliams